docs/proposals/device-state.md

```diff
@@ -0,0 +1,215 @@
+		DeviceName:      gs.DeviceName,
+		State:           states,
+		DeviceNamespace: gs.DeviceNamespace,
+	}
+
+	log.Printf("send statesRequest", statesRequest.DeviceName, statesRequest.State)
+	if err := grpcclient.ReportDeviceStates(statesRequest); err != nil {
+		klog.Errorf("fail to report device states of %s with err: %+v", gs.DeviceName, err)
+	}
+}
+```
+
+### Edgecore design
+- In the dmiserver module, implement ReportDeviceStates and send the received device states information to TwinGroup through beehive.
+- In TwinGroup, match the "/state/update" topic, send it to the device moudle, and execute the action DeviceStateUpdate.
+- In DeviceStateUpdate, update the device states information of the sqlite database and transmit it to the cloud.
+
+```golang
+func dealDeviceStateUpdate(context *dtcontext.DTContext, resource string, msg interface{}) error {
+	message, ok := msg.(*model.Message)
+	if !ok {
+		return errors.New("msg not Message type")
+	}
+
+	updatedDevice, err := dttype.UnmarshalDeviceUpdate(message.Content.([]byte))
+	if err != nil {
+		klog.Errorf("Unmarshal device info failed, err: %#v", err)
+		return err
+	}
+	deviceID := resource
+	defer context.Unlock(deviceID)
+	context.Lock(deviceID)
+	doc, docExist := context.DeviceList.Load(deviceID)
+	if !docExist {
+		return nil
+	}
+	device, ok := doc.(*dttype.Device)
+	if !ok {
+		return nil
+	}
+
+	// state refers to definition in mappers-go/pkg/common/const.go
+	state := strings.ToLower(updatedDevice.State)
+	switch state {
+	case "online", "offline", "ok", "unknown", "disconnected":
+	default:
+		return nil
+	}
+	var lastOnline string
+	if state == "online" || state == "ok" {
+		lastOnline = time.Now().Format("2006-01-02 15:04:05")
+	}
+	for i := 1; i <= dtcommon.RetryTimes; i++ {
+		err = dtclient.UpdateDeviceFields(
+			device.ID,
+			map[string]interface{}{
+				"last_online": lastOnline,
+				"state":       updatedDevice.State,
+			})
+		if err == nil {
+			break
+		}
+		time.Sleep(dtcommon.RetryInterval)
+	}
+	if err != nil {
+		return err
+	}
+	device.State = updatedDevice.State
+	device.LastOnline = lastOnline
+	payload, err := dttype.BuildDeviceState(dttype.BuildBaseMessage(), *device)
+	if err != nil {
+		return err
+	}
+	topic := dtcommon.DeviceETPrefix + device.ID + dtcommon.DeviceETStateUpdateResultSuffix
+	context.Send(device.ID,
+		dtcommon.SendToEdge,
+		dtcommon.CommModule,
+		context.BuildModelMessage(modules.BusGroup, "", topic, messagepkg.OperationPublish, payload))
+
+	msgResource := "device/" + device.ID + dtcommon.DeviceETStateUpdateSuffix
+	context.Send(deviceID,
+		dtcommon.SendToCloud,
+		dtcommon.CommModule,
+		context.BuildModelMessage("resource", "", msgResource, model.UpdateOperation, string(payload)))
+	return nil
+}
+```
+
+### Cloudcore design    
+- Add device states-related definitions to the crd definition of device.
+```yaml
+  last_online:
+    description: 'Optional: The last time the device was online.'
+    type: string
```

Comment:
There should be a `status` field in the crd of device-instance before. Can we add some content directly to this field?